civiform/civiform
# [Program Card Images] Show program image preview on admin program list page

### Description

This PR updates the "Program dashboard" page on the admin UI to show a small preview of the image currently associated with each program. If there's no image for the program, an empty rectangle with a gray border is shown. If there are different images in the Active version vs Draft version, those different images are also shown.

Most of this PR is moving the logic that renders the image into a `ProgramImageUtils` class so that it can be reused for both the applicant and the admin.

Most of the snapshot changes are because I've changed the program row UI to use `"w-1/4"` for the program title & description instead of `"w-1/3"`.  This gives more space for the image to show. I'll double-check with UX that this change is okay before submitting.

## Release notes

### Checklist

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)


#### User visible changes


### Instructions for manual testing

1. Enable the `PROGRAM_CARD_IMAGES` flag.
2. Verify that the program dashboard page shows empty rectangles for all programs, indicating that there are no images saved for the programs.
3. Add a program image for a program (Edit > Edit Program Image > Upload an image > Save)
4. Return to program dashboard page and verify that the image you uploaded is shown in the Draft row of the program.
5. Publish the program.
6. Verify that the image you uploaded is now shown in the Active row of the program.
7. Add a new program image for the same program.
4. Return to program dashboard page and verify that the old image is shown in the Active row and the new image is shown in the Draft row.

### Issue(s) this completes

Epic #5676

Fixes #6281  (sub-task)
